entities/BaseAddress.php

```diff
@@ -30,8 +40,14 @@ public function __construct(
          * 
          * @var int
          */
-        private int $id
+        private int $id,
+        Loyalty $loyalty
     ) {
+        $this->query = $loyalty->loyalTo(
+            $connection,
+            "addresses",
+            $id
+        );
```

Comment:
Code in constructor is a code smell, but it is justified because of the requirements towards the designed system. `ColumnQuery` implementation can be different, however, it must use the given connection and make queries to the specified record in a specified table.